File: Emulator/src/main/content/global/skill/summoning/familiar/npc/BullAntNPC.java
package content.global.skill.summoning.familiar.npc;

import content.global.skill.summoning.familiar.BurdenBeast;
import content.global.skill.summoning.familiar.Familiar;
import content.global.skill.summoning.familiar.FamiliarSpecial;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.game.node.entity.skill.Skills;
import core.game.world.update.flag.context.Animation;
import core.game.world.update.flag.context.Graphics;
import core.plugin.Initializable;

/**
 * The type Bull ant npc.
 */
@Initializable
public class BullAntNPC extends BurdenBeast {

    /**
     * Instantiates a new Bull ant npc.
     */
    public BullAntNPC() {
        this(null, 6867);
    }

    /**
     * Instantiates a new Bull ant npc.
     *
     * @param owner the owner
     * @param id    the id
     */
    public BullAntNPC(Player owner, int id) {
        super(owner, id, 3000, 12087, 12, 9, WeaponInterface.STYLE_CONTROLLED);
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new BullAntNPC(owner, id);
    }

    @Override
    protected boolean specialMove(FamiliarSpecial special) {
        if (owner.getSettings().getRunEnergy() >= 100) {
            owner.getPacketDispatch().sendMessage("You already have full run energy.");
            return false;
        }
        int amount = owner.getSkills().getStaticLevel(Skills.AGILITY) / 2;
        visualize(Animation.create(7896), Graphics.create(1382));
        owner.getSettings().updateRunEnergy(-amount);
        return true;
    }

    @Override
    public void visualizeSpecialMove() {
        owner.visualize(new Animation(7660), new Graphics(1296));
    }

    @Override
    public int[] getIds() {
        return new int[]{6867, 6868};
    }

}

File: Server/src/main/content/global/skill/summoning/familiar/npc/BullAntNPC.kt
package content.global.skill.summoning.familiar.npc

import content.global.skill.summoning.familiar.BurdenBeast
import content.global.skill.summoning.familiar.Familiar
import content.global.skill.summoning.familiar.FamiliarSpecial
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.game.node.entity.skill.Skills
import core.game.world.update.flag.context.Animation
import core.game.world.update.flag.context.Graphics
import core.plugin.Initializable
import shared.consts.Items
import shared.consts.NPCs

@Initializable
class BullAntNPC @JvmOverloads constructor(owner: Player? = null, id: Int = NPCs.BULL_ANT_6867) :
    BurdenBeast(owner, id, 3000, Items.BULL_ANT_POUCH_12087, 12, 9, WeaponInterface.STYLE_CONTROLLED) {
    override fun construct(owner: Player, id: Int): Familiar {
        return BullAntNPC(owner, id)
    }

    override fun specialMove(special: FamiliarSpecial): Boolean {
        if (owner.settings.runEnergy >= 100) {
            owner.packetDispatch.sendMessage("You already have full run energy.")
            return false
        }
        val amount = owner.getSkills().getStaticLevel(Skills.AGILITY) / 2
        visualize(Animation.create(7896), Graphics.create(1382))
        owner.settings.updateRunEnergy(-amount.toDouble())
        return true
    }

    override fun visualizeSpecialMove() {
        owner.visualize(Animation(7660), Graphics(1296))
    }

    override fun getIds(): IntArray {
        return intArrayOf(NPCs.BULL_ANT_6867, NPCs.BULL_ANT_6868)
    }
}
